jvb/src/main/java/org/jitsi/videobridge/octo/OctoEndpointMessageTransport.java

```diff
@@ -90,13 +104,26 @@ public BridgeChannelMessage addReceiver(@NotNull AddReceiverMessage message)
     public BridgeChannelMessage removeReceiver(@NotNull RemoveReceiverMessage message)
     {
         Conference conference = octoEndpoints.getConference();
-        AbstractEndpoint endpoint = conference.getEndpoint(message.getEndpointId());
-        // Since we currently broadcast everything in Octo, we may receive messages intended for another bridge. Handle
-        // only those that reference an endpoint local to this bridge.
-        if (endpoint instanceof Endpoint)
+        if (MultiStreamConfig.config.getEnabled())
         {
-            // TODO Implement source name based receiver constraints for Octo
-            endpoint.removeReceiver(message.getBridgeId());
+            String sourceName = message.getSourceName();
+            AbstractEndpoint endpoint = conference.findSourceOwner(sourceName);
+            // Since we currently broadcast everything in Octo, we may receive messages intended for another bridge.
+            // Handle only those that reference an endpoint local to this bridge.
+            if (endpoint instanceof Endpoint)
+            {
+                endpoint.removeReceiverV2(message.getBridgeId(), sourceName);
+            }
+        }
+        else
+        {
+            AbstractEndpoint endpoint = conference.getEndpoint(message.getEndpointId());
```

Comment:
I've been wondering why `conference.getLocalEndpoint` is not used here even though the comment talks about "endpoint local to this bridge"?

Reply:
No good reason as far as I can tell. The code might have been written before `getLocalEndpoint` was available